lib/remap-functions/src/parse_key_value.rs

```diff
@@ -0,0 +1,370 @@
+use nom::{
+    self,
+    branch::alt,
+    bytes::complete::{escaped, tag, take_until, take_while1},
+    character::complete::{char, satisfy, space0},
+    combinator::{map, rest},
+    multi::{many1, separated_list1},
+    sequence::{delimited, preceded, tuple},
+    IResult,
+};
+use remap::prelude::*;
+use std::collections::BTreeMap;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseKeyValue;
+
+impl Function for ParseKeyValue {
+    fn identifier(&self) -> &'static str {
+        "parse_key_value"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "key_value_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+            Parameter {
+                keyword: "field_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let key_value_delimiter = arguments.optional("key_value_delimiter").map(Expr::boxed);
+        let field_delimiter = arguments.optional("field_delimiter").map(Expr::boxed);
```

Comment:
Minor nit, but you can also set the default at compile-time, to remove the extra check in the expression itself.

E.g. if `optional` returns `None`, then set it to the literal expression `"="`.

This also makes `type_def` a bit easier to implement, as you no longer need `merge_optional` but only `merge` (the outcome is the same, since literals are infallible).